catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,363 @@
+import org.codice.ddf.catalog.async.data.ProcessCreateItemImpl;
+import org.codice.ddf.catalog.async.data.ProcessResourceImpl;
+import org.codice.ddf.catalog.async.data.ProcessUpdateItemImpl;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessCreateItem;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessDeleteItem;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessRequest;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResourceItem;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessUpdateItem;
+import org.codice.ddf.catalog.async.plugin.api.internal.PostProcessPlugin;
+import org.codice.imaging.nitf.core.common.NitfFormatException;
+import org.codice.imaging.nitf.fluent.NitfParserInputFlow;
+import org.codice.imaging.nitf.render.NitfRenderer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jaiimageio.jpeg2000.impl.J2KImageReaderSpi;
+import com.google.common.io.ByteSource;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.plugin.PluginExecutionException;
+import net.coobird.thumbnailator.Thumbnails;
+
+/**
+ * This pre-storage plugin creates and stores the NITF thumbnail and NITF overview images. The
+ * thumbnail is stored with the Metacard while the overview is stored in the content store.
+ */
+public class NitfPostProcessPlugin implements PostProcessPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NitfPostProcessPlugin.class);
+
+    private static final String IMAGE_JPEG = "image/jpeg";
+
+    private static final int THUMBNAIL_WIDTH = 200;
+
+    private static final int THUMBNAIL_HEIGHT = 200;
+
+    private static final String JPG = "jpg";
+
+    private static final String OVERVIEW = "overview";
+
+    private static final String ORIGINAL = "original";
+
+    private static final String DERIVED_IMAGE_FILENAME_PATTERN = "%s-%s.%s";
+
+    // non-word characters equivalent to [^a-zA-Z0-9_]
+    private static final String INVALID_FILENAME_CHARACTER_REGEX = "[\\W]";
+
+    private static final double DEFAULT_MAX_SIDE_LENGTH = 1024.0;
+
+    private double maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+
+    static {
+        IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
+    }
+
+    @Override
+    public ProcessRequest<ProcessCreateItem> processCreate(ProcessRequest<ProcessCreateItem> input)
+            throws PluginExecutionException {
+        return processProcessResourceRequest(input);
+    }
+
+    private <T extends ProcessResourceItem> ProcessRequest<T> processProcessResourceRequest(ProcessRequest<T> input) throws PluginExecutionException {
+        if (input == null) {
+            throw new PluginExecutionException("Unable to process null input.");
```

Comment:
nit: the same string literal is used multiple times here.  Maybe make it a constant.